metaflow/datastore/datastore_set.py

```diff
@@ -30,7 +30,7 @@ def __init__(self,
                                attempt=attempt,
                                event_logger=event_logger,
                                monitor=monitor,
-                               data_obj=json.loads(data_blob),
+                               data_obj=json.loads(isinstance(data_blob,str) and data_blob or data_blob.decode('utf-8')),
```

Comment:
Could you verify what happens if we always `data_blob.decode('utf-8')`? If it doesn't we can also leverage the `util` module in Metaflow which simplifies this task for you. 

When `data_obj=None` it seems we are leveraging `load_metadata` which does the decoding unconditionally [1], so we can probably do the same here, since the `data_blob` is directly returned from S3 client `get_many`.

[1] https://github.com/Netflix/metaflow/blob/master/metaflow/datastore/s3.py#L184

Reply:
Yeah my 1st version of the PR was to always decode but it crashes in 3.6  since in python > 3.5 the data_blob is already a binary and there's no decode method on that type..

I'm not sure why the data_blob type is of different type between python versions, something to investigate... ideas welcome there.

I'll checkout the util module... 

Note I'll be away for the holidays so it may take a while for me to take a second look...

Reply:
Thanks for the super quick response.

Regarding the util library, what I have in mind is something like `json.loads(util.to_unicode(data_blob))`

I find the current conditional slightly harder to reason about, so seeing if an existing function can handle the compatibility issues across (py) versions.

Also - no rush on this PR; since we anyway plan to merge only when we have a critical mass of work for the next release. Happy holidays!

Reply:
OK cool yeah that would surely be better, I also wasn't found of that conditional...

I'll test it out..

Note also, the current test framework doesn't test for 3.5, I usually like to provide a test with a fix, let me know if that's possible to do or if it's better to wait...

Reply:
@seeravikiran changed as per review

Reply:
Sorry I don't remember how this slipped my radar; but your new change looks good to me.

I'll approve once I can test the change myself. Resolving this conversation.